### ISSUE TITLE
Hitting NotFoundGraphError2: Graph.addDirectedEdge: source node error in CI

### ISSUE THREAD

**Issue by tranhl** [git-town/action#42]
Original issue: https://github.com/git-town/git-town/issues/4195 by @EngHabu

---

We use git-town actions to populate the stack view in the PR description.
It has been working well until recently (today?) where we started hitting this:

```
/home/runner/work/_actions/git-town/action/v1/dist/index.js:24877
        throw new NotFoundGraphError("Graph.".concat(name, ': source node "').concat(source, '" not found.'));
              ^

NotFoundGraphError2: Graph.addDirectedEdge: source node "<a branch name here>" not found.
    at addEdge (/home/runner/work/_actions/git-town/action/v1/dist/index.js:24877:15)
    at Graph.<computed> [as addDirectedEdge] (/home/runner/work/_actions/git-town/action/v1/dist/index.js:26139:20)
    at /home/runner/work/_actions/git-town/action/v1/dist/index.js:42861:15
    at Array.forEach (<anonymous>)
    at main (/home/runner/work/_actions/git-town/action/v1/dist/index.js:42860:[16](https://github.com/unionai/cloud/actions/runs/11693203423/job/32564253997?pr=9227#step:3:22))
    at run (/home/runner/work/_actions/git-town/action/v1/dist/index.js:47005:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v20.13.1
```

It's unclear to me what's going on... The referenced branch is not related to my PR and isn't part of the stack...

**Comment by tranhl** [comment]
@EngHabu I'll need a bit more information from you to figure out what's going on. Would you be able to share the action's inputs from the problematic workflow run? You can find this information from the step's output:
![image](https://github.com/user-attachments/assets/9376ab86-8032-4789-9c6a-eecc6454106e)

I've seen this error before, I believe it could be related to #36. Once you provide me with some further information I'll dig into the issue ASAP!

**Comment by mwisner** [comment]
We're seeing the same thing also,

```
/home/runner/work/_actions/git-town/action/v1/dist/index.js:24877
          throw new NotFoundGraphError("Graph.".concat(name, ': source node "').concat(source, '" not found.'));
                ^
  
  NotFoundGraphError2: Graph.addDirectedEdge: source node "..." not found.
      at addEdge (/home/runner/work/_actions/git-town/action/v1/dist/index.js:24877:[15](https://github.com/stockx/iron/actions/runs/12011258930/job/33481569022#step:3:19))
      at Graph.<computed> [as addDirectedEdge] (/home/runner/work/_actions/git-town/action/v1/dist/index.js:26139:20)
      at /home/runner/work/_actions/git-town/action/v1/dist/index.js:42861:15
      at Array.forEach (<anonymous>)
      at main (/home/runner/work/_actions/git-town/action/v1/dist/index.js:42860:[16](https://github.com/stockx/iron/actions/runs/12011258930/job/33481569022#step:3:20))
      at run (/home/runner/work/_actions/git-town/action/v1/dist/index.js:47005:10)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  
  Node.js v20.13.1
```

I'm not sure what information I can provide other than:
- We have a lot of branches (800+) I've deleted a bunch but the action still stops at the same branch.
- No forks
- Copy/paste the action out of the example without any changes
- Branch failing on does exist

**Comment by stephenwade** [comment]
@mwisner Would you be able to share the action's inputs from the problematic workflow run? You can get this from the step's output, as seen in https://github.com/git-town/action/issues/42#issuecomment-2475023158.

**Comment by tranhl** [comment]
@mwisner @EngHabu

I've released [v1.0.6](https://github.com/git-town/action/releases/tag/v1.0.6) which contains fixes for #19 & #36. I believe this should resolve the error! Please monitor the action & report back if the issue persists.

I'll be closing this issue just to keep things tidy, but if the error comes up again I'm more than happy to reopen it 😊

**Comment by mwisner** [comment]
Works great, thank you!!!
